src/java.base/share/classes/java/util/Properties.java

```diff
@@ -833,6 +842,13 @@ public void save(OutputStream out, String comments)  {
      * After the entries have been written, the output stream is flushed.
      * The output stream remains open after this method returns.
      *
+     * @implNote This method invokes the {@link #entrySet()} method
+     * and writes out the returned key and element pairs
+     * in the natural sort order of those keys. If subclasses override
+     * the {@code entrySet} method and return a different {@code Set} instance,
+     * then the property list is written out in the iteration order of
+     * that returned {@code Set}
```

Comment:
Rewording a bit:

"The keys and elements are written in the natural sort order of the keys in the `Properties.entrySet` unless `entrySet` is overridden by a subclass to return a different instance."

"different instance" is a bit hard to implement given that entrySet() returns a new synchronized set each time.
typo: missing final "."

Reply:
yes - maybe we could work on the wording. Perhaps:
> The keys and elements are written in the natural sort order of the keys in the Properties.entrySet unless entrySet is overridden by a subclass to return a different set implementation.